### ISSUE TITLE
DatabasesInParallel

### ISSUE THREAD

**Issue by ghost** [olahallengren/sql-server-maintenance-solution#687]
**Description of the issue**
We have (8) TLOG backup jobs running across hundreds of databases in parallel.
If one of the backups is still running when the job executes in 20 minutes none of the databases are backed up.

**SQL Server version and edition**
SQL 2016 13.0.7016.1

**Version of the script**
2022-01-02 13:58:13

**What command are you executing?**
DatabaseBackup LOG DatabasesInParallel

**What output are you getting?**
Nothing. The other (7) jobs just close because the one very large database is still doing a large LOG backup.

**Comment by bwiggins10** [comment]
Hey Keith,
A few questions here:

What's the specific parameters of your 8 log backup jobs?
What's the schedule of these / how frequently are you running log backups?
How large is the database in question?

Other than seeing if something is abnormal with the above, I'd say to make 1 specific log backup job for that very large database, and make the other separate 8 jobs exclude that database with something such as the following:
@Databases = 'USER_DATABASES, -VeryLargeDatabaseNameHere',

**Comment by KeithCorser** [comment]
Ben,

Here are your answers you requested:

1.)  Specific parameters =   " sqlcmd -E -S $(ESCAPE_SQUOTE(SRVR)) -d master -Q "EXECUTE [dbo].[DatabaseBackup] @Databases = 'USER_DATABASES', @Directory = N'\\acespbak1\SQLBackups\SQLTLogBackups', @BackupType = 'LOG', @Verify = 'N', @CleanupTime = 168, @CheckSum = 'Y', @maxtransfersize= 65537, @DatabasesInParallel = 'Y', @LogToTable = 'Y'" -b"

2.) LOG backups run every 20 minutes.

3.) Data file = 614400.00 mb
   Log file = 409728.68 mb

4.) One off solutions like mentioned below got us into this mess. Evidently someone excluded the largest database from it's own backup job back in February.

Everything is caught up after this weekend but thought you should be aware that in the current design none of the LOG backups run again until all of them are done.


Thanks,
Keith


________________________________

**Comment by bwiggins10** [comment]
Yeah in this case, there's definitely that known issue with the @DatabasesInParallel parameter but given the inactivity of updates to this repo, I'd recommend in the meantime having that secondary job for that large databases.

Definitely good to have monitoring in place, however aggressive/frequent that works, but something that checks for log backups for databases in full recovery model and alerts if there's no log backup in X time, so that if someone had made an accidental exclusion or disabled the job, you'd be notified about it within X time. Especially if there's a lot of transactions for that database, leaving it without log backups can blow up its log and nobody wants that.

One other thing I'd recommend is increasing the frequency of the log backups in addition to moving that large database to its own job. There's a really good BO article I've linked below on this topic, but it's actually less overhead to do them more frequently if your system can handle it with the amount of databases.

[https://www.brentozar.com/archive/2014/02/back-transaction-logs-every-minute-yes-really/](url)

**Comment by KeithCorser** [comment]
Ben.

We put our own modification in to the sp to correct the behavior. Just wanted to let others know the issue was there. 🙂
For monitoring of backups I dug up some policies I had created in the past to make sure backups are being done going forward.



Thanks,
Keith
________________________________

**Comment by bwiggins10** [comment]
Glad to hear it! 😄 
Hoping this post brings some attention to OLA to give us some of that much needed update love.

**Comment by olahallengren** [comment]
This is the current design. I agree that for transaction log backups it is not optimal.

I will look into how it can be improved.
